lib/src/reduction_tree.dart

```diff
@@ -13,9 +13,11 @@ import 'package:meta/meta.dart';
 import 'package:rohd/rohd.dart';
 import 'package:rohd_hcl/rohd_hcl.dart';
 
-/// A generator which constructs a tree of radix-input / 1-output modules.
+/// Recursive Node for Reduction Tree
 class ReductionTree extends Module {
-  /// The radix-sized input operation to be performed at each node.
+  /// Operation to be performed at each node. Note that [operation] can widen
+  /// the output. The logic function must support the operation for [2 to radix]
```

Comment:
nit: `[2 to radix]` is not a link to anything, maybe also worth clarifying that this means any number from 2 to [radix].